### PR TITLE
Create owner model

### LINKED ISSUE #49: Create owner model

### PR DESCRIPTION
**Pull request by davidgohzk**
Fixes #49

### CODE REVIEW

File: src/main/java/seedu/address/model/owner/Address.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     * @param address A valid address.
+     */
+    public Address(String address) {
+        requireNonNull(address);
+        checkArgument(isValidAddress(address), MESSAGE_CONSTRAINTS);
+        value = address;
+    }
+
+    /**
+     * Returns true if a given string is a valid email.
```
Comment (jloh02) [inline src/main/java/seedu/address/model/owner/Address.java:34]: Do you mean to be a valid landed address? not an email?

File: src/main/java/seedu/address/model/owner/Email.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        + "    - have each domain label consist of alphanumeric characters, separated only by hyphens, if any.";
+    // alphanumeric and special characters
+    private static final String ALPHANUMERIC_NO_UNDERSCORE = "[^\\W_]+"; // alphanumeric characters except underscore
+    private static final String LOCAL_PART_REGEX = "^" + ALPHANUMERIC_NO_UNDERSCORE + "([" + SPECIAL_CHARACTERS + "]"
+        + ALPHANUMERIC_NO_UNDERSCORE + ")*";
+    private static final String DOMAIN_PART_REGEX = ALPHANUMERIC_NO_UNDERSCORE
+        + "(-" + ALPHANUMERIC_NO_UNDERSCORE + ")*";
+    private static final String DOMAIN_LAST_PART_REGEX = "(" + DOMAIN_PART_REGEX + "){2,}$"; // At least two chars
+    private static final String DOMAIN_REGEX = "(" + DOMAIN_PART_REGEX + "\\.)*" + DOMAIN_LAST_PART_REGEX;
+    public static final String VALIDATION_REGEX = LOCAL_PART_REGEX + "@" + DOMAIN_REGEX;
```
Comment (jloh02) [inline src/main/java/seedu/address/model/owner/Email.java:32]: Thanks for making the validation regex clear by splitting into multiple regexes!

File: src/test/java/seedu/address/model/owner/EmailTest.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        assertTrue(Email.isValidEmail("PeterJack+1190@example.com")); // '+' symbol in local part
+        assertTrue(Email.isValidEmail("PeterJack-1190@example.com")); // hyphen in local part
+        assertTrue(Email.isValidEmail("a@bc")); // minimal
+        assertTrue(Email.isValidEmail("test@localhost")); // alphabets only
+        assertTrue(Email.isValidEmail("123@145")); // numeric local part and domain name
+        assertTrue(Email.isValidEmail("a1+be.d@example1.com")); // mixture of alphanumeric and special characters
+        assertTrue(Email.isValidEmail("peter_jack@very-very-very-long-example.com")); // long domain name
+        assertTrue(Email.isValidEmail("if.you.dream.it_you.can.do.it@example.com")); // long local part
+        assertTrue(Email.isValidEmail("e1234567@u.nus.edu")); // more than one period in domain
+    }
```
Comment (jloh02) [inline src/test/java/seedu/address/model/owner/EmailTest.java:68]: Good test coverage of various edge cases!

File: src/main/java/seedu/address/model/owner/Phone.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * Represents a Owner's phone number in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidPhone(String)}
+ */
+public class Phone {
+
+
+    public static final String MESSAGE_CONSTRAINTS =
+        "Phone numbers should only contain numbers, and it should be at least 3 digits long";
```
Comment (jloh02) [inline src/main/java/seedu/address/model/owner/Phone.java:14]: Perhaps we should also add in that no spaces or + should be added to be clear

File: src/test/java/seedu/address/model/owner/PhoneTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        // null phone number
+        assertThrows(NullPointerException.class, () -> Phone.isValidPhone(null));
+
+        // invalid phone numbers
+        assertFalse(Phone.isValidPhone("")); // empty string
+        assertFalse(Phone.isValidPhone(" ")); // spaces only
+        assertFalse(Phone.isValidPhone("91")); // less than 3 numbers
+        assertFalse(Phone.isValidPhone("phone")); // non-numeric
+        assertFalse(Phone.isValidPhone("9011p041")); // alphabets within digits
+        assertFalse(Phone.isValidPhone("9312 1534")); // spaces within digits
```
Comment (jloh02) [inline src/test/java/seedu/address/model/owner/PhoneTest.java:33]: Add a test case to test for with country extension e.g. "+198765432"

**Review by jloh02 — CHANGES_REQUESTED** [review]
Thanks for the work! Just a few minor changes

**Review by naythee169 — COMMENTED** [review]
LGTM

**Review by jloh02 — APPROVED** [review]
